pyramid_zipkin/request_helper.py

```diff
@@ -53,14 +37,36 @@ def get_trace_id(request):
     completely random trace id.
 
     :param: current active pyramid request
-    :returns: string representaiton of zipkin trace id
+    :returns: a 64-bit hex string
     """
     if 'X-B3-TraceId' in request.headers:
-        return request.headers['X-B3-TraceId']
+        id = request.headers['X-B3-TraceId']
     elif 'zipkin.trace_id_generator' in request.registry.settings:
-        return request.registry.settings['zipkin.trace_id_generator'](request)
+        id = request.registry.settings['zipkin.trace_id_generator'](request)
     else:
-        return thrift_compatible_string(generate_random_64bit_string())
+        id = generate_random_64bit_string()
+
+    # Backwards compatibility for <=v0.8.1
+    # If the trace id is a hex value that starts with '0x' or '-0x',
+    # convert to the unsigned form before proceeding.
+    if id.startswith('0x') or id.startswith('-0x'):
+        id = _signed_hex_to_unsigned_hex(id)
+    id = id.zfill(16)
+
+    assert _is_hex_string(id) and len(id) == 16
+    return id
+
+
+def _is_hex_string(s):
+    try:
+        int(s, 16)
+        return True
+    except ValueError:
+        return False
+
+
+def _signed_hex_to_unsigned_hex(s):
+    return '{0:x}'.format(struct.unpack('Q', struct.pack('q', int(s, 16)))[0])
```

Comment:
Would be great to give an example of the type of conversion this function is meant to apply.